pipeline/thumbnails.go

```diff
@@ -0,0 +1,93 @@
+package pipeline
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/livepeer/catalyst-api/clients"
+	"github.com/livepeer/go-tools/drivers"
+)
+
+const framesEverySecs = 5
+
+func GenerateThumbs(input *url.URL, output *url.URL) error {
+	in, err := clients.SignURL(input)
+	if err != nil {
+		return err
+	}
+	tempDir, err := os.MkdirTemp(os.TempDir(), "thumbs-*")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDir)
+
+	args := []string{
+		"-i", in,
+		"-vf", "select=eq(pict_type\\,I)",
+		"-vsync", "vfr",
+		"-vf", fmt.Sprintf("fps=1/%d", framesEverySecs),
+		path.Join(tempDir, "/keyframes_%03d.jpg"),
+	}
+
+	timeout, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(timeout, "ffmpeg", args...)
+
+	var outputBuf bytes.Buffer
+	var stdErr bytes.Buffer
+	cmd.Stdout = &outputBuf
+	cmd.Stderr = &stdErr
+
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("error running ffmpeg [%s] [%s] %w", outputBuf.String(), stdErr.String(), err)
+	}
+
+	// generate the webvtt file
+	files, err := filepath.Glob(path.Join(tempDir, "keyframes*"))
+	if err != nil {
+		return err
+	}
+	builder := bytes.Buffer{}
+	_, err = builder.WriteString("WEBVTT\n")
+	if err != nil {
+		return err
+	}
+	timestamp := time.Time{}
+	outputLocation := output.JoinPath("thumbnails").String()
+	for _, file := range files {
```

Comment:
Same here - a function like `GenerateVTT(filename string)` in `video/thumbs.go` could potentially generate and write the file to disk and once it returns, this function can call `clients.UploadToOSURL` to prevent recursive dependency in video package. It also makes the code cleaner and abstracts away all video-y logic away from the pipeline package. wdyt?